### PR TITLE
Fix exact alarm permission on Android 14

### LINKED ISSUE #293: Gotify requires pernissions to display push notifications
Hello, gotify app and server have been working for quite some ti e now but recently, the app hasn't been working. When I launch the app I get a message of gotify requires permissions which I've enabled everything I can think of but won't let me move forward .
I can register the server, it creates the client but then gives me the permissions error.

Could this have anything to do with android beta? I'm running upsidedowncake UPB2.230407.014

Hope you can help.

### PR DESCRIPTION
**Pull request by cyb3rko**
I've now started working on it and it seems that it's functional, at least on Android 14.  
Feel free to test it out, before merging I'd like to verify that everything's working fine on Android 13 and below.

Little insight on why it's not working atm: The permission for scheduling exact alarms is not to be handled like other permissions via a normal dialog, rather you have to send an Intent to the settings page, where you can enable the schedule permission for that specific app.

Closes #293

### CODE REVIEW

File: app/src/main/kotlin/com/github/gotify/init/InitializationActivity.kt [hunk 1]
```diff
@@ -55,7 +71,14 @@ internal class InitializationActivity : AppCompatActivity() {
...
             runWithNeededPermissions {
-                tryAuthenticate()
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S) {
+                    val manager = ContextCompat.getSystemService(this, AlarmManager::class.java)
+                    if (manager?.canScheduleExactAlarms() == false) {
+                        alarmDialog()
+                    } else {
+                        tryAuthenticate()
+                    }
+                }
```
Comment (jmattheis) [inline app/src/main/kotlin/com/github/gotify/init/InitializationActivity.kt:81]: This looks a little suspicious, if the token exists and the SDK_INT is < VERSION_CODES.S, then nothing will be done, previously tryAuthenticate was called. Haven't tested this PR yet.

File: app/src/main/kotlin/com/github/gotify/init/InitializationActivity.kt [hunk 2]
```diff
@@ -34,6 +39,17 @@ internal class InitializationActivity : AppCompatActivity() {
...
 
+    @RequiresApi(Build.VERSION_CODES.S)
+    private val activityResultLauncher =
+        registerForActivityResult(StartActivityForResult()) {
+            val manager = ContextCompat.getSystemService(this, AlarmManager::class.java)
+            if (manager?.canScheduleExactAlarms() == true) {
+                tryAuthenticate()
+            } else {
+                alarmDialog()
+            }
```
Comment (jmattheis) [inline app/src/main/kotlin/com/github/gotify/init/InitializationActivity.kt:50]: Can this part be extracted to another separate method and reused on the lines 75+?
The checks are a little different == false and == true, but it looks doable.

**Review by jmattheis — COMMENTED** [review]
I'll test this PR in the following week.

File: app/src/main/kotlin/com/github/gotify/init/InitializationActivity.kt (hunk 1 above)
Comment (cyb3rko) [reply app/src/main/kotlin/com/github/gotify/init/InitializationActivity.kt:81]: True. That's exactly why I wanted to test it on other SDKs.  
I rather speeded the implementation :|

File: app/src/main/kotlin/com/github/gotify/init/InitializationActivity.kt (hunk 2 above)
Comment (cyb3rko) [reply app/src/main/kotlin/com/github/gotify/init/InitializationActivity.kt:50]: Doable

**Comment by cyb3rko** [comment]
Verified that it's functional on both Android 14 and Android <13 now.

File: app/src/main/kotlin/com/github/gotify/init/InitializationActivity.kt [hunk 3]
```diff
@@ -1,12 +1,17 @@
 package com.github.gotify.init
```
Comment (jmattheis) [inline app/src/main/kotlin/com/github/gotify/init/InitializationActivity.kt:2]: Sorry for taking so long to review this (:. I'm having trouble getting it to work on Android 12/API 31 and Android 12l/API 32. With the current state of master it works fine, but with these changes it crashes after the login, or shows a blank page:
[android12.webm](https://github.com/gotify/android/assets/14895212/11bd004a-8e44-4a02-b0e3-b159c553f9f1)

API 30,33,34 works fine. There is no exception logged, after pressing login & creating the client.

Comment (cyb3rko) [reply app/src/main/kotlin/com/github/gotify/init/InitializationActivity.kt:2]: Interesting, sorry for the problems.
I will take a look as soon as possible.

Comment (cyb3rko) [reply app/src/main/kotlin/com/github/gotify/init/InitializationActivity.kt:2]: I think I've found my error.  
Works now on my emulators on SDK 25, 31, 34.

Comment (jmattheis) [reply app/src/main/kotlin/com/github/gotify/init/InitializationActivity.kt:2]: Nice, I've tested it with SDK 23,30,31,32,33 & 34 and it works fine. Thanks for debugging and fixing this (:. The dependency problem was also fixed somehow, so I'll don't change the logging dependency yet.

**Review by jmattheis — APPROVED** [review]

**Pull request merged**